common/src/test/java/io/netty/util/HashedWheelTimerTest.java

```diff
@@ -154,4 +146,72 @@ public void run(final Timeout timeout) throws Exception {
 
         timer.stop();
     }
+
+    @Test
+    public void testPendingTimeoutCountBeforeAndAfterTimeoutIsScheduledAndExecuted() throws InterruptedException {
+        HashedWheelTimer timer = new HashedWheelTimer(100, TimeUnit.MILLISECONDS);
+        timer.newTimeout(createNoOpTimerTask(), 1, TimeUnit.SECONDS);
+        long pendingTimeoutsCount = timer.pendingTimeoutsCount();
+        assertTrue("Pending timeouts count not 1 after adding 1 timeout. Actual count = "
+            + pendingTimeoutsCount, pendingTimeoutsCount == 1);
+
+        Thread.sleep(200); // Wait for next tick. This should remove timeoutToCancel1
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Remove this use of "Thread.sleep()". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS2925)